### PR TITLE
DS-1779 publish with GitHub actions

### PR DESCRIPTION
**Pull request by kkanishkan**
Adds functionality to use GitHub actions to create a release when a tag is pushed.

### CODE REVIEW

File: .github/workflows/releaseOnVersionTag.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Publish Release
+on: 
+  push:
+    tags:
+      - 'v*'
+jobs:
+  release:
+    runs-on: ubuntu-latest
```
Comment (Tarun-S) [inline .github/workflows/releaseOnVersionTag.yml:8]: Should we use our vendored Amazon ECR docker image here? Not sure if it's easy to do, but it looks like it's possible: 

https://stackoverflow.com/questions/64033686/how-can-i-use-private-docker-image-in-github-actions

Comment (dangreenisrael) [reply .github/workflows/releaseOnVersionTag.yml:8]: This is a public repo, we are not going to want to reference our private ECR images

File: .github/workflows/releaseOnVersionTag.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    steps:
+      - name: Create Release
+        uses: Roang-zero1/github-create-release-action@master
```
Comment (dangreenisrael) [inline .github/workflows/releaseOnVersionTag.yml:11]: Can we do this without relying on this library?

Comment (dangreenisrael) [reply .github/workflows/releaseOnVersionTag.yml:11]: We may be able to do what poetry themselves do here:

https://github.com/python-poetry/poetry/blob/master/.github/workflows/release.yml#L235-L255

Comment (kkanishkan) [reply .github/workflows/releaseOnVersionTag.yml:11]: I can try to make my own Github action package as well

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@v2
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.9"
+      - name: Install poetry
+        run: |
+          pip install poetry
+      - name: Build package
```
Comment (dangreenisrael) [inline .github/workflows/release.yml:19]: I think we need a `poetry install` step here.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        run: |
+          poetry build
+      - name: Publish
+        run: |
+          poetry publish --username=__token__ --password='${{ secrets.PYPI_TOKEN}}'
```
Comment (dangreenisrael) [inline .github/workflows/release.yml:24]: I've added the token to the repo's secrets

File: .github/workflows/release.yml (hunk 3 above)
Comment (kkanishkan) [reply .github/workflows/release.yml:22]: Added

**Review by dangreenisrael — APPROVED** [review]

**Pull request merged**